### PR TITLE
Simplified creation of k8s secrets for IPSec integration 

### PR DESCRIPTION
**Pull request by aanm**
read per commit message

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/7048)

### CODE REVIEW

**Comment by aanm** [comment]
test-me-please

**Comment by jrfastab** [comment]
test-missed-k8s

**Comment by jrfastab** [comment]
test-missed-k8s

**Comment by jrfastab** [comment]
test-me-please

**Comment by aanm** [comment]
test-missed-k8s

**Comment by jrfastab** [comment]
hit FQDN failure it seems unlikely that this could cause any issues there so lets retry.

**Comment by jrfastab** [comment]
test-missed-k8s

**Review by ianvernon — APPROVED** [review]
LGTM for docs.

File: pkg/datapath/linux/node.go [hunk 1]
```diff
@@ -651,7 +650,10 @@ func (n *linuxNodeHandler) replaceNodeIPSecOutRoute(ip *net.IPNet) {
 		}
 	}
 
-	route.Upsert(n.createNodeIPSecOutRoute(ip), n.nodeConfig.MtuConfig)
+	_, err := route.Upsert(n.createNodeIPSecOutRoute(ip), n.nodeConfig.MtuConfig)
+	if err != nil {
+		log.WithError(err).Error("Unable to replace node ip sec in-route")
```
Comment (tgraf) [inline pkg/datapath/linux/node.go]: I don't understand this error message.

```Unable to replace IPsec node outgoing route```

File: pkg/datapath/linux/node.go [hunk 2]
```diff
@@ -671,7 +673,10 @@ func (n *linuxNodeHandler) replaceNodeIPSecInRoute(ip *net.IPNet) {
 		}
 	}
 
-	route.Upsert(n.createNodeIPSecInRoute(ip), n.nodeConfig.MtuConfig)
+	_, err := route.Upsert(n.createNodeIPSecInRoute(ip), n.nodeConfig.MtuConfig)
+	if err != nil {
+		log.WithError(err).Error("Unable to replace node ip sec in-route")
```
Comment (tgraf) [inline pkg/datapath/linux/node.go]: Same here

File: pkg/datapath/linux/ipsec/ipsec_linux.go [hunk 3]
```diff
@@ -287,8 +293,20 @@ func LoadIPSecKeysFile(path string) error {
 			return fmt.Errorf("missing IPSec keys or invalid format")
 		}
 
-		authname, authkey := s[0], []byte(s[1])
-		encname, enckey := s[2], []byte(s[3])
+		// As we have released the v1.4.0 docs telling the users to write the
```
Comment (tgraf) [inline pkg/datapath/linux/ipsec/ipsec_linux.go]: Move this into the function where the prefix is removed

**Review by tgraf — CHANGES_REQUESTED** [review]

**Comment by aanm** [comment]
test-me-please

**Review by jrfastab — APPROVED** [review]

**Comment by aanm** [comment]
hit https://github.com/cilium/cilium/issues/6786 in the CI, merging as it was the only test failure and it's a known issue.

**Pull request merged**